apps/veyfi/utils/index.ts

```diff
@@ -16,3 +17,39 @@ export function getVotingPower(lockAmount: BigNumber, unlockTime: TMilliseconds)
 	}
 	return lockAmount.div(MAX_LOCK).mul(duration);
 }
+
+export const keyBy = <T1, T2 extends keyof T1 & string>(array: T1[], key: T2): TDict<T1 | undefined> => 
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	(array || []).reduce((r, x): TDict<T1> => ({...r, [(x as any)[key]]: x}), {});
```

Comment:
Wouldn't `x[key]` be `T2` here? Which in turn is a `string`?
```suggestion
	(array || []).reduce((r, x): TDict<T1> => ({...r, [x[key] as T2]: x}), {});
```

Reply:
I still don't know why you need the ` as unknown`, thinking it must be your TypeScript version.

That's not required for me, can just cast as string directly